src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.cs

```diff
@@ -691,11 +691,6 @@ internal void EnsureStoreIsValid()
             if (Disposed)
                 throw new ObjectDisposedException(null, SR.IsolatedStorage_StoreNotOpen);
 
-            if (IsDeleted)
-            {
-                throw new IsolatedStorageException(SR.IsolatedStorage_StoreNotOpen);
-            }
-
             if (_closed)
```

Comment:
Oops, I think you're right. If the public DeleteDirectory() method is called, we delete the directory so IsDeleted will be true but we don't Close() and set _closed.
Perhaps then we should keep the IsDeleted check but change it to throw InvalidOperationException like desktop. ie 
```c#
            if (_closed || IsDeleted)
                throw new InvalidOperationException(SR.IsolatedStorage_StoreNotOpen);
```